lib/common/issues/src/dashboard.rs

```diff
@@ -0,0 +1,177 @@
+use std::collections::HashMap;
+use std::sync::{Mutex, MutexGuard, OnceLock};
+
+use serde::{Serialize, Serializer};
+
+use crate::issue::{CodeType, Issue, IssueRecord};
+
+#[derive(Default, Serialize)]
+struct Dashboard {
+    pub issues: HashMap<CodeType, IssueRecord>,
+}
+
+impl Dashboard {
+    /// Activates an issue, returning true if the issue was not active before
+    fn add_issue(&mut self, issue: impl Issue) -> bool {
+        let issue = IssueRecord::from(issue);
+        self.issues.insert(issue.code.clone(), issue).is_none()
+    }
+
+    /// Deactivates an issue by its code, returning true if the issue was active before
+    fn remove_issue<S: AsRef<str>>(&mut self, code: S) -> bool {
+        self.issues.remove(code.as_ref()).is_some()
+    }
+}
+
+fn dashboard<'a>() -> MutexGuard<'a, Dashboard> {
+    static DASHBOARD: OnceLock<Mutex<Dashboard>> = OnceLock::new();
+    match DASHBOARD
+        .get_or_init(|| Mutex::new(Dashboard::default()))
```

Comment:
It seems like a `RwLock` could be a fit as a few operations only need read access.
I'd recommend using directly the variant from `parking_lot`.

Reply:
I hadn't considered this, and now I want to change the access pattern for insertion, for example, because the same issue is possible to be generated many times. So that we can check for existence only with a read before acquiring write access

Reply:
I you want to avoid juggling with locks you could try to use a concurrent hashmap like https://github.com/xacrimon/dashmap